presentation/src/main/java/me/androidbox/presentation/HomeActivity.kt

```diff
@@ -3,20 +3,47 @@ package me.androidbox.presentation
 import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
+import androidx.activity.viewModels
+import androidx.compose.runtime.collectAsState
+import androidx.core.splashscreen.SplashScreen.Companion.installSplashScreen
 import androidx.navigation.compose.rememberNavController
 import dagger.hilt.android.AndroidEntryPoint
+import me.androidbox.domain.authentication.ResponseState
+import me.androidbox.presentation.login.viewmodel.SplashScreenViewModel
 import me.androidbox.presentation.navigation.NavigationGraph
+import me.androidbox.presentation.navigation.Screen
 import me.androidbox.presentation.ui.theme.BusbyTaskyTheme
 
 @AndroidEntryPoint
 class HomeActivity : ComponentActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        val splashScreenViewModel by viewModels<SplashScreenViewModel>()
+
+        installSplashScreen().apply {
+            this.setKeepOnScreenCondition {
+                splashScreenViewModel.isCompletedState.value
+            }
+        }
+
+        splashScreenViewModel.authenticateUser()
+
         setContent {
+            val authenticatedState = splashScreenViewModel.authenticationState.collectAsState()
+            val destination = when(authenticatedState.value) {
+                is ResponseState.Success -> {
+                    /* TODO Go to the Agenda Screen when implemented */
+                    Screen.RegisterScreen.route
+                }
+                else -> {
+                    Screen.LoginScreen.route
+                }
+            }
```

Comment:
@philipplackner and @MKiperszmid Is this the best way to use this splashscreen? Everything works as expected. But I just want to make sure.

Reply:
Looks good, yes